src/frame/headers.rs

```diff
@@ -343,6 +342,17 @@ impl<T> From<Headers> for Frame<T> {
     }
 }
 
+impl fmt::Debug for Headers {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("Headers")
+            .field("stream_id", &self.stream_id)
+            .field("stream_dep", &self.stream_dep)
+            .field("flags", &self.flags)
+            // `fields` and `pseudo` purposefully not included
```

Comment:
I lean towards yes, personally. But I'm not an immovable object on this, either.

For debugging that wants to see the actual bytes, an IO layer that wraps and logs bytes read and written would allow seeing everything, wouldn't it? At least then, it's explicit that you're dumping all the bytes, instead of how easy it is for someone to just enable trace logging entirely for their application.